app/event-creator.tsx

```diff
@@ -1,14 +1,57 @@
 'use client'
 
-import React from "react";
+import React, { useState } from "react";
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
 
 export default function EventCreator() {
 
     const [name, setName] = React.useState("");
     const [date, setDate] = React.useState("");
     const [calendarId, setCalendarId] = React.useState("");
 
+    function validateInput() {
+
+        var valid = true;
+        var errstyle = "2px solid red";
+        var initstyle = "1px solid grey";
+
+        if (name.length == 0) {
+            const el = document.getElementById("name");
+            el.style.border = errstyle;
+
+            valid = false;
+            el.onchange = () => { el.style.border = initstyle, name => setName(name)};
+        }
+
+        if (date == null || date.length == 0) {
+            const el = document.getElementById("date");
+            el.style.border = errstyle;
+
+            valid = false;
+            el.onclick = () => { el.style.border = initstyle, date => setDate(date) };
+        }
+
+        if (calendarId.length == 0) {
+            const el = document.getElementById("calendar");
+            el.style.border = errstyle;
+
+            valid = false;
+            el.onchange = () => { el.style.border = initstyle, calendarId => setCalendarId(calendarId) };
+        }
+
+        if (!valid) {
+            alert('Invalid Input')
+            return;
+        }
+
+        if (valid) {
```

Comment:
Is this if really needed? If the input is invalid you show an alert and return. Could this be written as:
`if(valid) {
    handleClick();
    return:
}

alert('Invalid Input')
`

Reply:
i know about this, but I left it for clarity :)